mythril/laser/ethereum/state.py

```diff
@@ -1,16 +1,62 @@
-from z3 import BitVec, BitVecVal, Solver, ExprRef, sat
+from z3 import BitVec, BitVecVal, BitVecRef, BitVecNumRef, BitVecSort, Solver, ExprRef, Concat, sat, simplify, Array
+from z3.z3types import Z3Exception
 from mythril.disassembler.disassembly import Disassembly
 from copy import copy, deepcopy
 from enum import Enum
 from random import randint
 
 from mythril.laser.ethereum.evm_exceptions import StackOverflowException, StackUnderflowException
 
-
 class CalldataType(Enum):
     CONCRETE = 1
     SYMBOLIC = 2
 
+class Calldata:
+    def __init__(self, tx_id, starting_calldata: bytes=None):
+        self.tx_id = tx_id
+        self._calldata = Array('{}_calldata'.format(self.tx_id), BitVecSort(256), BitVecSort(8))
+        self.starting_calldata = starting_calldata if starting_calldata else []
+
+    def set_global_state(self, state):
+        self.state = state
+        for i in range(len(self.starting_calldata)):
+            self.state.mstate.constraints.append(self._calldata[BitVecVal(i, 256)] == BitVecVal(self.starting_calldata[i], 8))
```

Comment:
I think calldata is typically list of symbols and integers, so this might fail in some circumstances when we are trying to convert a BitVecVal to a symbol.